### DISCUSSION TITLE
Firmware upload failed

### DISCUSSION THREAD

**Discussion by dac2020** [arendst/Tasmota#16923] [General]
What can I do when I cannot even upload the minimal firmware to a device?
I get upload failed, not enough space which explains what the problem is but what can I do about it?

I thought the minimal build would fit on pretty much anything?
In this case, it's the ESP32R4 V2, RocketController relay.

Is there some way of clearing some space? Are logs kept on the device perhaps that I could delete?
![image](https://user-images.githubusercontent.com/9727841/198095319-aabd43e9-8c9b-436c-9940-d35def804dd2.png)

**Comment by sfromis** [comment]
If you try uploading minimal, you are doing something very wrong. That is not relevant for ESP32. What exact upgrade are you trying to do, and exactly how?

**Reply by dac2020** [reply]
I'm starting to get confused. 
I was told when I started that when there isn't enough room, use the minimal build that Tasmota provides, then upgrade from there.
This is what I've done for every device I have but now this one won't update so I posted.

Not sure what I could be doing wrong since I'm doing the same as I've done since starting.

**Reply by barbudor** [reply]
minimal is only for esp8266
You should have no size problem on the ESP32
Which binary are you trying to flash ? A ready binary ? from where ? 
Or a binary that you built yourself ? With which configuration ?

**Reply by sfromis** [reply]
You ignored my questions about details....

**Reply by dac2020** [reply]
I am very sure I used that minimal on all of the devices that I have so far and didn't have any problems until now.
I think I got that from the resulting default build when using https://github.com/arendst/Tasmota.git.
So, maybe it was generated when I ran  ./compiler.sh tasmota.

Not really sure at this point to be honest and would have to dig through my many posts to find it.

**Reply by sfromis** [reply]
You likely used minimal on ESP8266 devices, not ESP32 devices where it is just not part of how Tasmota works. Since you still are not telling us what you were doing (outside of using minimal), it is hard to guess what you did wrong (other than using minimal).

**Reply by dac2020** [reply]
I didn't ignore it, just didn't notice what I've missed and still don't.

This?
>What exact upgrade are you trying to do, and exactly how?

I mentioned the device, rocketcontroller, 4 relay device, ESP32R4 V2. 
I shared the chip and the details of the device in terms of space.
I'm using the firmware upgrade option  from the main menu.

What did I forget to include? Why I guess?

Because I wanted to re-flash the device to a base Tasmota to see what the default rules are since I didn't make a copy before I removed the rules and there were some. Now even after running 'rule 0 "', there are still three rules and I wanted to better understand why. 

This is posted in my other question where I need to review the replies :).
https://github.com/arendst/Tasmota/discussions/16760

**Reply by barbudor** [reply]
Base firmware for this ESP32 based firmware is `tasmota32.bin`
There is no rules by default in any official tasmota binary
beside upgrading will just change the firmware but will keep existing settings as they are
unless you issue a `reset 1` command that will wipe everything out and restore the default tasmota settings which means, you will have to reset everything from the beginning : wifi, mqtt if any, and the device template because official tasmota32.binary doesn't know anything about RocketController until you have set the template.

**Reply by barbudor** [reply]
Which means that if your device came with rules that were set by the manufacturer, you won't find them by loading official tasmota32 firmware on it (neither the template) that I hope is the 1st thing that you saved before playing with the device
Hopefully you would find it here : https://templates.blakadder.com/robotdyn_ESP32R4.html

**Reply by sfromis** [reply]
I asked for **exact** details, not a general abstract description of what you want to do. Details like name/size of the binary, how specifically you try to upload it, and what log messages you get when you try to do it.

**Reply by dac2020** [reply]
Wow, you don't have to scold me, I'm not a child playing games with you. 
You know I'm just learning here and you also know I've been making progress so please be a little more patient.

I didn't know about the settings remaining. I would have thought and did that when you write new firmware, it overwrites everything. 

I do have the template, the device is working fine. There is nothing wrong with it. 

No general abstract, I also explained why. I wanted to write a basic firmware to it to see if those rules would change. 
I removed them using the command suggested but as mentioned, continued to see three rules so wondered why.

As I said, I think the file was generated when I used the compiler.
The file is called tasmota-minimal.bin and is 376016 in size. I might have renamed it to minimal, I'm not sure. 
I could run the compiler again and see if it generates it again.

As I said I used the firmware update page so not sure what else to add on that.

I had not looked at the log so I'll do it now. Web log is set to 3.

00:13:19.591 HTP: Firmware Upgrade
00:13:25.729 UPP: Multicast disabled
00:13:25.968 CFG: Saved, Count 219, Bytes 4096
00:13:25.970 UPL: File tasmota-minimal.bin
00:13:25.994 UPL: freespace=1900544 filesize=376016
00:13:25.000 UPL: File type 1
00:13:27.742 UPL: Progress 100 KB
00:13:29.464 UPL: Progress 200 KB
00:13:31.001 UPL: Progress 300 KB
00:13:32.098 UPP: Multicast (re)joined
00:13:32.118 HTP: Upload done
00:13:38.248 HTP: Main Menu

![image](https://user-images.githubusercontent.com/9727841/198156136-f3753b5a-44da-4f49-a371-48d3139478e6.png)

00:13:43.995 HTP: Consoles
00:13:45.973 HTP: Console

**Reply by sfromis** [reply]
Well, once more, using `tasmota-minimal.bin` is **not relevant** for ESP32. Do not do that.

I was assuming (hoping) that you only started to try with `tasmota-minimal.bin` after having problems uploading the binary you want to upgrade to, and that problem was what I was asking about. If you had not gotten to that state, it was because you should never have used `tasmota-minimal.bin` for ESP32.

If that `tasmota-minimal.bin` was generated when **trying** to build for ESP32, you were not successful in building for ESP32 instead of ESP8266. All standard build binary names for ESP32 start with `tasmota32`, when it is only `tasmota`, it is for ESP8266.

**Reply by dac2020** [reply]
I have a thousand notes since starting this and I do recall someone mentioning there is a difference between the chips but I didn't remember that the firmware files are specific, like openwrt for example. 

So this seems to be where I'm at.
I have an ESP32 build which means I have to rename that so I don't forget and I need to now generate a new build for the ESP8266.

**Reply by barbudor** [reply]
An ESP32 build should always start by "tasmota32-someting.bin"
For more specifics with ESP32 variants, ESP32-C3 should be named tasmota32c3-something.bin, ESP32-S3 would be tasmota32s2-something.bin etc ...
And there is no .gz for ESP32 and its variants

ESP32 variants are very different one from the others. Some are even using a totally different CPU core. So each should be considered as a totally different processor

tasmota-something.bin(.gz) is reserved of ESP8266/ESP8285 (the later being en ESP8266 with a build-in 1MB flash)
